builds/Makefile

```diff
@@ -27,132 +36,136 @@ downloads/$(MONO_IOS_FILENAME) downloads/$(MONO_MAC_FILENAME):
-# so disable 'futimens' and 'utimensat'
-
-COMMON_ACVARS =  \
-	ac_cv_func_fstatat=no \
-	ac_cv_func_readlinkat=no \
-	ac_cv_func_futimens=no \
-	ac_cv_func_utimensat=no \
-
-COMMON_LDFLAGS=-Wl,-no_weak_imports
-
-BITCODE_CFLAGS=-fexceptions
-BITCODE_LDFLAGS=-framework CoreFoundation -lobjc -lc++
-BITCODE_CONFIGURE_FLAGS=--enable-llvm-runtime --with-bitcode=yes
-
-ifdef ENABLE_BITCODE_ON_IOS
-IOS_BITCODE_CFLAGS=$(BITCODE_CFLAGS) -fembed-bitcode-marker
-IOS_BITCODE_CXXFLAGS=$(BITCODE_CFLAGS) -fembed-bitcode-marker
-IOS_BITCODE_LDFLAGS=$(BITCODE_LDFLAGS)
-IOS_BITCODE_CONFIGURE_FLAGS=$(BITCODE_CONFIGURE_FLAGS)
-endif
-
-#
-# The install-* targets should not depend on any of the build-* targets,
-# because it will slow down the install targets unnecessarily (the build-*
-# targets will recurse into multiple directories, which would be wasteful
-# since in most cases the build has already been done)
-#
-# Currently 'make install' takes about 4 seconds (without running any build-*
-# targets)
-#
-# This means that 'make all install' will not work in this directory, it must
-# be 'make all && make install'.
-#
-
-# This turns off the clang: warning: argument unused during compilation
-# warnings when using clang+ccache together
-CCACHE_CFLAGS=-Qunused-arguments
-CCACHE_CXXFLAGS=-Qunused-arguments
-
 #
 # Configuration for the mono sdk makefiles
 #
-SDK_CONFIG=$(MONO_PATH)/sdks/Make.config
 SDK_ARGS= \
 	XCODE_DIR=$(XCODE_DEVELOPER_ROOT) XCODE32_DIR=$(XCODE94_DEVELOPER_ROOT) \
 	IOS_VERSION=$(IOS_SDK_VERSION) IOS_VERSION_MIN=$(MIN_IOS_SDK_VERSION) \
 	TVOS_VERSION=$(TVOS_SDK_VERSION) TVOS_VERSION_MIN=$(MIN_TVOS_SDK_VERSION) \
 	WATCHOS_VERSION=$(WATCH_SDK_VERSION) WATCHOS_VERSION_MIN=$(MIN_WATCHOS_SDK_VERSION) \
 	IGNORE_PROVISION_LLVM=1
 
-ifdef DISABLE_DOWNLOAD_LLVM
-SDK_ARGS += DISABLE_DOWNLOAD_LLVM=1
-endif
-
+SDK_CONFIG = $(MONO_PATH)/sdks/Make.config
 SDK_BUILDDIR = $(MONO_PATH)/sdks/builds
-SDK_DESTDIR = $(abspath $(MONO_PATH)/sdks/out)
 
-$(MONO_PATH)/sdks/Make.config:
+$(SDK_CONFIG):
 	echo "DISABLE_ANDROID=1" > $@
+	echo "DISABLE_DESKTOP=1" >> $@
+	echo "DISABLE_WASM=1" >> $@
 
-##
-## Xamarin.Mac
-##
+ifdef DISABLE_BUILDS_MAKEFILE_DEP
+BUILDS_MAKEFILE_DEP =
+else
+BUILDS_MAKEFILE_DEP = Makefile
+endif
+
+.stamp-configure-mono: $(SDK_CONFIG)
+	$(MAKE) -C $(SDK_BUILDDIR) configure-ios configure-mac $(SDK_ARGS)
+	$(Q) touch $@
+
+.stamp-compile-mono: .stamp-configure-mono $(MONO_DEPENDENCIES) $(BUILDS_MAKEFILE_DEP) $(TOP)/Make.config
+	$(MAKE) -C $(SDK_BUILDDIR) package-ios package-ios-bcl package-mac package-mac-bcl $(SDK_ARGS)
+	$(Q) touch $@
+
+clean-local::
+	$(MAKE) -C $(SDK_BUILDDIR) clean $(SDK_ARGS)
+	$(Q) rm -f .stamp-compile-mono
+
+all-local:: .stamp-compile-mono .stamp-mono-ios-sdk-destdir
+
+endif
+
+.stamp-mono-ios-sdk-destdir: .stamp-$(MONO_BUILD_MODE)
+	ln -sf $(MONO_IOS_SDK_DESTDIR) mono-ios-sdk-destdir
+	$(Q) touch $@
+
+ifdef FALSE  # TODO see if we still need this?
```

Comment:
If "we're done", do we need this?

Reply:
Ah yeah, good catch. I discussed with Miguel and he said nobody except VSMac (using system mono) relies on the mono-extensions bits so this can go away here.